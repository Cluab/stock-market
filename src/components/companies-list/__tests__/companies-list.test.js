import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';
import { configureStore } from '@reduxjs/toolkit';
import '@testing-library/jest-dom/extend-expect';
import CompaniesList from '../companies-list';
import valueReducer from '../../../redux/companies/companies';

const stock = [[{
  symbol: 'JPM',
  date: '2021-12-31',
  revenue: 121685000000,
  costOfRevenue: 0,
  costAndExpenses: 0,
  ebitda: 73047000000,
  grossProfit: 0,
  grossProfitRatio: 0,
  operatingExpenses: 0,
  operatingIncome: 0,
  eps: 15.39,
  operatingIncomeRatio: 0,
}],
[{
  symbol: 'JPM',
  date: '2021-12-31',
  revenue: 121685000000,
  costOfRevenue: 0,
  costAndExpenses: 0,
  ebitda: 73047000000,
  grossProfit: 0,
  grossProfitRatio: 0,
  operatingExpenses: 0,
  operatingIncome: 0,
  eps: 15.39,
  operatingIncomeRatio: 0,
}]];

const store = configureStore({ reducer: { stock: valueReducer } });

describe('companies component test cases', () => {
  test('should match with snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <CompaniesList
              props={stock}
            />
          </Router>
        </Provider>,
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match with input data ', () => {
    render(
      <Provider store={store}>
        <Router>
          <CompaniesList
            props={stock}
          />
        </Router>
      </Provider>,
    );
    const stockSymbol = screen.getByTestId('test-symbol').textContent;
    const stockRevenue = screen.getByTestId('test-revenue').textContent;

    expect(stockSymbol).toBe(String(stock[0][0].symbol));
    expect(stockRevenue).toBe(String(stock[0][0].revenue));
  });
});
