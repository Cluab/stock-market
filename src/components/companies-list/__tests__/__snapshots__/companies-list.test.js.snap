// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`companies component test cases should match with snapshot 1`] = `
Array [
  <a
    href="/details/JPM"
    onClick={[Function]}
  >
    <div
      className="main-comp-section"
    >
      <span
        className="fa-solid fa-money-bill-trend-up"
      />
      <div
        className="main-comp-name"
      >
        <h2>
          JPM
        </h2>
        <span>
          121685000000
          / Revenue
        </span>
      </div>
    </div>
  </a>,
  <div
    className="main-page-bar"
  >
    <p>
      US STOCK MARKET
    </p>
  </div>,
  <div
    className="comp-name-list"
  >
    <a
      href="/details/JPM"
      onClick={[Function]}
    >
      <div>
        <span
          className="fa-solid arrow-stock fa-circle-arrow-right"
        />
        <i
          className="fa-solid fa-money-bill-trend-up"
        />
        <h3
          data-testid="test-symbol"
        >
          JPM
        </h3>
        <span
          data-testid="test-revenue"
        >
          121685000000
        </span>
      </div>
    </a>
  </div>,
]
`;
